### ISSUE TITLE
[BUG]Multi currency error

### ISSUE THREAD

**Issue by amitsaxena01** [frappe/erpnext#15398]
If we create a Purchase Order > Purchase Invoice > Journal Entry for the payment for say a payment of $1000 at Rs66 to 1 USD; The ledger shows a value which is different from $1000. It seems that it is converting $1000 to INR in Journal Entry and then from Indian Rupees to USD in ledger and the rate of conversion is not being taken from the value entered in the Journal Entry but from somewhere else.

-Amit

**Comment by FinForceConsulting** [comment]
We are facing a similar issue in
ERPNext: v11.0.3-beta.32 () (staging)
Frappe Framework: v11.0.3-beta.46 () (staging)

The steps I have taken are:
1. Setup the exchange rate of USD to TZS at 2312
2. Created SI for a customer in USD for $35. On the transaction doc, it shows TZS amount = 80,920
3. Post submit of SI, I check the GL report for the accounting entries.

I see transaction correctly in TZS = 80920, but I see the transaction in USD at a different value, which is not $35. However, surprisingly same transaction reviewed after few hours shows things alright.

I wonder what might be causing this issue of reflecting different rates of USD in the GL reports

**Comment by meaziz** [comment]
Any updates or guidance about this issue,
We are facing it now, And we are probably ready to fix it.

**Comment by meaziz** [comment]
Well this is a little bit strange,
I managed to get it working locally,

But when I tried it in production environment I found it was not updating the View.
But worked after rebooting the server.
My guess is that the Code Caches the Conversion Exchange rate and serves it accordingly.

So you will only face problems If you view the General Ledger Report.
then try applying the currency Filter on it.
then found out that It didn't fetch an appropriate exchange rate.
so you decide to go in put a manual record with the exchange rate, date.
then nothing happen; the rate doesn't update accordingly,
Only after a server restart it does.
